File: poc-kronos-frontend/src/model/avaliacao/ResultadoHabilidade.js
class ResultadoHabilidade {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idAvaliado = dados ? dados.idAvaliado : null;
    this.idAvaliacaoHabilidade = dados ? dados.idAvaliacaoHabilidade : null;
    this.nota = dados ? dados.nota : null;
    this.idMencao = dados ? dados.idMencao : null;
    this.descarte = dados ? dados.descarte : null;
    this.motivoDescarte = dados ? dados.motivoDescarte : null;
    this.idsAvaliados = dados ? dados.idsAvaliados : [];
    this.idsMencoes = dados ? dados.idsMencoes : [];
  }
  static validations(required) {
    return {
      idAvaliacaoHabilidade: { required },
      idMencao: { required }
    };
  }

  static validationMessages() {
    return {
      idAvaliacaoHabilidade: {
        required: "Avaliação e Habilidade é obrigatória"
      },
      idMencao: {
        required: "Menção é obrigatória"
      }
    };
  }
}
export default ResultadoHabilidade;

File: src/main/java/br/com/kronos/backend/aplicacao/resultado/ResultadoHabilidade.java

package br.com.kronos.backend.aplicacao.resultado;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import lombok.Builder;
import lombok.Getter;

@Getter
public class ResultadoHabilidade {
    
    private Long id;
    private Long idAvaliado;
    private Long idAvaliacaoHabilidade;
    private Double nota;
    private Long idMencao; 
    private boolean descarte; 
    private String motivoDescarte; 
  
	@Builder
    public ResultadoHabilidade(Long id, Long idAvaliado, Long idAvaliacaoHabilidade, Double nota, 
                         Long idMencao, boolean descarte, String motivoDescarte  ) {

		super();
        this.id = id;
        this.idAvaliado = exigirNaoNulo(idAvaliado, "Avaliado"); 
        this.idAvaliacaoHabilidade = exigirNaoNulo(idAvaliacaoHabilidade, "Avaliação e habilidade");    
        this.nota = nota; 
        this.idMencao = idMencao; 
        this.descarte = exigirNaoNulo(descarte, "Descarte"); 
        this.motivoDescarte = motivoDescarte; 
	}
}